pyuvdata/uvflag/uvflag.py

```diff
@@ -1166,6 +1176,30 @@ def set_telescope_params(self, overwrite=False, warn=True):
             # ensure that the Nants_telescope comes first so shapes work out below
             telescope_params.remove("_Nants_telescope")
             telescope_params.insert(0, "_Nants_telescope")
+
+            set_ant_metadata = True
+            if self.type != "waterfall" and "_antenna_numbers" in telescope_params:
+                # need to check that all antennas on the object are in the telescope's
+                # antenna_numbers
+                if self.type == "antenna":
+                    ants_to_check = self.ant_array
+                else:
+                    ants_to_check = np.union1d(self.ant_1_array, self.ant_2_array)
+
+                if not all(
+                    ant in telescope_obj.antenna_numbers for ant in ants_to_check
+                ):
+                    warnings.warn(
+                        "Not all antennas with data have metadata in the telescope "
```

Comment:
This function is only called if one or more of telescope_location, antenna_names, antenna_numbers or antenna_positions is missing from the file. So this should only be called for old files.

If you have some of the antenna metadata but not the other parts (this would be very strange -- there's not a way for this to happen through the normal file writing mechanism) or if you deliberately call this function separately from the file read, then this check happens.